src/cli/__tests__/telemetry.test.ts

```diff
@@ -0,0 +1,118 @@
+import assert from 'node:assert';
+import { describe, it } from 'node:test';
+
+import { createReporter, detectCI, parseDsn, parseFrames } from '../telemetry.ts';
+
+describe('telemetry', () => {
+  describe('detectCI', () => {
+    it('detects GitHub Actions', () => {
+      assert.deepStrictEqual(
+        detectCI({
+          GITHUB_ACTIONS: 'true',
+        }),
+        'github',
+      );
+    });
+
+    it('detects CircleCI', () => {
+      assert.deepStrictEqual(
+        detectCI({
+          CIRCLECI: 'true',
+        }),
+        'circleci',
+      );
+    });
+
+    it('detects Travis CI', () => {
+      assert.deepStrictEqual(
+        detectCI({
+          TRAVIS: 'true',
+        }),
+        'travis',
+      );
+    });
+
+    it('detects Azure Pipelines', () => {
+      assert.deepStrictEqual(
+        detectCI({
+          TF_BUILD: 'true',
+        }),
+        'azure',
+      );
+    });
+  });
+
+  describe('parseDsn', () => {
+    it('parses a DSN', () => {
+      const dsn = 'https://public-key@host/project-id';
+      const parsed = parseDsn(dsn);
+      assert.deepStrictEqual(parsed, {
+        host: 'host',
+        projectId: 'project-id',
+        key: 'public-key',
+        protocol: 'https',
+      });
+    });
+  });
+
+  describe('parseFrames', () => {
+    it('returns an array of Sentry frames', async () => {
+      const stack = `Error: testing
+    at funcName (file:///Users/username/path/file.js:10:5)
+    at bound (node:foo/bar:433:15)
+    at funcName2 (file:///Users/username/path/file2.js:20:10)
+    at funcName3 (file:///Users/username/path/file3.js:30:15)`;
+
+      const frames = await parseFrames(stack);
+
+      assert.deepStrictEqual(frames, [
+        {
+          function: 'funcName3',
+          raw_function: 'funcName3',
+          abs_path: '/Users/username/path/file3.js',
```

Comment:
Should we clean out the absolute path? Seems like maybe it could be a privacy issue?

Reply:
Yeah, maybe we can make it relative to pwd. What do you think?